app/Rules/StatusCode404.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Rules;
+
+use App\Services\UrlFetcher;
+use GuzzleHttp\Exception\BadResponseException;
+
+class StatusCode404 extends Rule
+{
+    protected $statusCode;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check()
+    {
+        $uri = rtrim($this->url, '/').'/page-to-test-404-response';
```

Comment:
What would you suggest? We just need a route that doesn't usually exist. So we could make it more unlikely (`/__page-to-test-404-response-for-kupo`) or actually random (`'/page-to-test-404-response-for-kupo-'Str::random(12)`), but I wanted to make it clear in the logs that this isn't an actual 404 from a visitor. And random pages would clutter the logs more probably.